src/JsonApi/Serializer/ItemNormalizer.php

```diff
@@ -37,16 +39,15 @@
  */
 final class ItemNormalizer extends AbstractItemNormalizer
 {
+    use ClassInfoTrait;
+
     const FORMAT = 'jsonapi';
 
     private $componentsCache = [];
-    private $resourceMetadataFactory;
 
-    public function __construct(PropertyNameCollectionFactoryInterface $propertyNameCollectionFactory, PropertyMetadataFactoryInterface $propertyMetadataFactory, IriConverterInterface $iriConverter, ResourceClassResolverInterface $resourceClassResolver, PropertyAccessorInterface $propertyAccessor = null, NameConverterInterface $nameConverter = null, ResourceMetadataFactoryInterface $resourceMetadataFactory, array $defaultContext = [])
+    public function __construct(PropertyNameCollectionFactoryInterface $propertyNameCollectionFactory, PropertyMetadataFactoryInterface $propertyMetadataFactory, IriConverterInterface $iriConverter, ResourceClassResolverInterface $resourceClassResolver, PropertyAccessorInterface $propertyAccessor = null, NameConverterInterface $nameConverter = null, ResourceMetadataFactoryInterface $resourceMetadataFactory, array $defaultContext = [], bool $allowUnmappedClass = false, DataTransformerInterface $dataTransformer = null)
```

Comment:
`$allowUnmappedClass` should be the latest parameter to create a better upgrade path